src/libvirtApi/common.js

```diff
@@ -315,14 +290,9 @@ function startEventMonitorDomains(connectionName) {
 
             switch (eventType) {
             case domainEvent.Defined:
-                domainGet({ connectionName, id:objPath });
-                break;
-
             case domainEvent.Undefined:
-                domainEventUndefined(connectionName, objPath);
-                break;
-
             case domainEvent.Started:
+            case domainEvent.Stopped:
```

Comment:
Because of this change, Stopped event now doesn't call `getVm` with `updateOnly = true`, which results in `updateOrAddVm` being called instead of `updateVm`. 
The difference this makes is that, as seen [here](https://github.com/cockpit-project/cockpit-machines/blob/main/src/reducers.js#L231), `updateVm` calls `clearUsageData`, which `updateOrAddVm` doesn't call. That leaves us with a bug that when VM is stopped (e.g. by clicking on "Force shutoff", it leaves metrics in "Usage" stuck:

![Screenshot from 2022-01-21 12-17-30](https://user-images.githubusercontent.com/42733240/150518271-2fbd1a12-daed-42a0-9654-ae220422df5b.png)

If we want to bring this change, we should also call `clearUsageData` for `updateOrAddVm` too.

Reply:
PS: It seems usage metrics clear after few moments anyway, but that is probably caused by different getVm refresh, so it's not really a big bug ;)

Reply:
Thanks for spotting this! I can reproduce. So this would already happen in main in some cases, when the recently fixed race condition would hit which got fixed in commit a724e2e29c9. So indeed it seems to make sense to do that in UPDATE_OR_ADD as well.

Fixed. But please double-check my logic, I'm not that well versed with the Redux stuff yet.